src/main/java/SubwayNetwork.java

```diff
@@ -0,0 +1,76 @@
+import com.sun.org.apache.bcel.internal.generic.FSUB;
+import jdk.internal.jline.internal.TestAccessible;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class SubwayNetwork {
+
+    private Map<Integer, SubwayStation.Station> stations;
+
+    public List<SubwayStation.Station> findShortestPath(SubwayStation.Station origin, SubwayStation.Station destination) throws IOException {
+        List<SubwayStation.Station> visitedStations = dijkstraAlgorithm(origin, destination);
+        return tracePath(visitedStations, origin, destination);
+    }
+
+    private List<SubwayStation.Station> dijkstraAlgorithm (SubwayStation.Station origin, SubwayStation.Station destination) throws IOException {
+        Converter converter = new Converter();
```

Comment:
In order to get the Stations to pass into this method, you'd already had to have a Converter.

You are reading the files multiple times by doing this. You aren't allowed to do that.